### PR TITLE
resource_manager: add metrics for avaiable RU

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?
see title
image

Issue Number: ref #4399

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/resourcemanager/server/metrics.go [hunk 1]
```diff
@@ -96,6 +96,14 @@ var (
 			Name:      "request_count",
 			Help:      "The number of read/write requests for all resource groups.",
 		}, []string{resourceGroupNameLabel, typeLabel})
+
+	availableRUCounter = prometheus.NewGaugeVec(
```
Comment (HuSharp) [inline pkg/mcs/resourcemanager/server/metrics.go:100]: Do we need to delete in `backgroundMetricsFlush`?

File: pkg/mcs/resourcemanager/server/manager.go [hunk 2]
```diff
@@ -358,6 +361,15 @@ func (m *Manager) backgroundMetricsFlush(ctx context.Context) {
 					delete(m.consumptionRecord, name)
 				}
 			}
+		case <-availableRUTicker.C:
+			m.RLock()
+			for name, group := range m.groups {
+				if name == reservedDefaultGroupName {
+					continue
+				}
+				availableRUCounter.WithLabelValues(name).Set(group.getRUToken())
```
Comment (nolouch) [inline pkg/mcs/resourcemanager/server/manager.go:370]: I think if ru <0, can let the metrics see it as 0?

File: pkg/mcs/resourcemanager/server/manager.go [hunk 3]
```diff
@@ -342,7 +345,7 @@ func (m *Manager) backgroundMetricsFlush(ctx context.Context) {
 
 			m.consumptionRecord[name] = time.Now()
 
-		case <-ticker.C:
+		case <-cleanUpTicker.C:
```
Comment (HuSharp) [inline pkg/mcs/resourcemanager/server/manager.go:348]: Do we need to cleanup not available resource group?

Comment (CabinfeverB) [reply pkg/mcs/resourcemanager/server/manager.go:348]: What does unavailable resource group mean?

Comment (HuSharp) [reply pkg/mcs/resourcemanager/server/manager.go:348]: When a group is recorded and then the user deletes it, but it is still in availableRUCounter.

Comment (CabinfeverB) [reply pkg/mcs/resourcemanager/server/manager.go:348]: I don't think we should delete it? We can avoid unnecessary computing and gRPC overhead by removing resource groups from the client side timely. Regular deletion seems to be ok here.

Comment (HuSharp) [reply pkg/mcs/resourcemanager/server/manager.go:348]: can delete with cleanUpTicker Regularly

File: pkg/mcs/resourcemanager/server/metrics.go [hunk 4]
```diff
@@ -96,6 +96,14 @@ var (
...
+		prometheus.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: ruSubsystem,
+			Name:      "available",
```
Comment (nolouch) [inline pkg/mcs/resourcemanager/server/metrics.go:104]:
```suggestion
			Name:      "available_ru",
```

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by nolouch — APPROVED** [review]
lgtm

**Review by HuSharp — APPROVED** [review]

**Pull request merged**